### ISSUE TITLE
Assert when allowFlowReversal=false

### ISSUE THREAD

**Issue by Mathadon** [iea-annex60/modelica-annex60#469]
When allowFlowReversal = false we assume a certain flow direction and simplify the equations accordingly. This assumption is however not always checked with an assert. I propose to add asserts for this since it may lead to nasty 'bugs', which may undermine the credibility of the library. 
However this would mean that there would be very many of these asserts, which increases computation time, but I think it's worth it.

**Comment by mwetter** [comment]
The fluid ports already declare the min and max values for `m_flow`. Can you achieve the above by enabling a flag in the simulator that checks for these values?
If this does not work, we can try using assertions and compare the translation and computing time. I think it is only at a few base classes where they would have to be added.

**Comment by Mathadon** [comment]
Dymola contains the following option:
```
Boolean Advanced.AssertAllInsideMinMax = false "Generate asserts that all variables are inside min/max";
```
the problem with this is that
1) (I think) this check does not allow to define a certain slack on the constraint
2) it's disabled by default, so Annex 60 users may be unaware of it

**Comment by Mathadon** [comment]
This is also linked to #531

**Comment by mwetter** [comment]
Closed as it is handled in #531.
